internal/controller/controllers/infraenv_controller.go

```diff
@@ -203,7 +203,6 @@ func (r *InfraEnvReconciler) ensureISO(ctx context.Context, log logrus.FieldLogg
 	infraEnv.Status.AgentLabelSelector = metav1.LabelSelector{MatchLabels: map[string]string{aiv1beta1.InfraEnvNameLabel: infraEnv.Name}}
 	var inventoryErr error
 	var Requeue bool
-	var updatedCluster *common.Cluster
 
 	kubeKey := types.NamespacedName{
```

Comment:
cluster deployment should not be required so i'm suggesting to open small PR that will make it optional in the CRD and then make all this part depend on if it's set or not.